Implement partner feedback on measles model

apulsipher opened the pull request:
- Add references to schools in measles model
- Add take home messages displayed prominently in value boxes
- Improve color scheme of charts and value boxes
- Add acknowledgement of partners and their logos 
- Add tooltips to each model control with additional details

apulsipher committed:
Add references to schools in measles model

apulsipher committed:
Add take home messages in colored boxes at top of model

apulsipher committed:
Improve color scheme of epicurve chart in measles model

apulsipher committed:
Swap colors of value boxes in measles model

apulsipher committed:
Add logos and acknowledgements of UDHHS and ForeSITE

apulsipher committed:
Add tooltips with details of each control parameter in the measles model

apulsipher committed:
Move images to assets/ folder

apulsipher committed:
Implement final tweaks from partner feedback

apulsipher committed:
Update NEWS.md

gvegayon commented:
This is what we are reviewing jointly, right?

apulsipher commented:
> This is what we are reviewing jointly, right?

Yes, that's the idea anyway.

damontoth commented on inst/models/shiny_measles.R:
```diff
@@ -315,49 +316,83 @@ measles_panel <- function(model_alt) {
+        step    = 1
+      ),
+      placement = "right",
+      "# of cases at the start of the simulation"
```
```suggestion
      "# of students infected with measles at the start of the simulation"
```

damontoth commented on inst/models/shiny_measles.R:
```diff
@@ -315,49 +316,83 @@ measles_panel <- function(model_alt) {
+            input_label = "quarantine_willingness"
+          ),
+          placement = "right",
+          "How willing infected people are to quarantine (1 = 100% willing, 0 = 0% willing)"
```
Quarantined people are not necessarily infected.
```suggestion
          "How willing people are to stay home from school when asked to quarantine (1 = 100% willing, 0 = 0% willing)"
```

damontoth commented on inst/models/shiny_measles.R:
```diff
@@ -315,49 +316,83 @@ measles_panel <- function(model_alt) {
+            step    = 1
+          ),
+          placement = "right",
+          "# of days an infected person is quarantined. 21 days is the CDC recommendation for measles quarantine."
```
```suggestion
          "# of days after potential exposure a quarantined person will stay home from school, if willing. 21 days is the CDC recommendation for measles quarantine."
```

damontoth commented on inst/models/shiny_measles.R:
```diff
@@ -367,69 +402,121 @@ measles_panel <- function(model_alt) {
+          slider_input_rate(
+            "measles", "Transmission probability", "0.99", input_label = "transmission_rate"),
+          placement = "right",
+          "The change an infected individual transmits the disease to a susceptible individual per day of the simulation"
```
```suggestion
          "The chance an infected individual transmits the disease to a contacted susceptible individual per day of the simulation"
```

damontoth commented on inst/models/shiny_measles.R:
```diff
@@ -367,69 +402,121 @@ measles_panel <- function(model_alt) {
+          slider_input_rate(
+            "measles", "Recovery probability (daily)", "0.14", input_label = "recovery_rate"),
+          placement = "right",
+          "The probability of an infected individual recovering per day of the simulation"
```
This should be more specific, but I'm not sure how it's used in the model. Is recovery_rate applied only after the end of the rash period? Also, it's a bit confusing to have this expressed as a rate, while the other stages below are expressed as (average?) durations. Does it make sense to express this as a duration (7 days = 1/0.14), i.e. "average days between end of rash and recovery" (if that's accurate).

gvegayon replied:
Actually, this is legacy code from a different model! We are not using it. I'll remove this entry from the program. Thanks for catching that, @damontoth!

gvegayon replied:
Yes, average, again (responding backward, sorry!)

damontoth commented on inst/models/shiny_measles.R:
```diff
@@ -367,69 +402,121 @@ measles_panel <- function(model_alt) {
+            step    = 1
+          ),
+          placement = "right",
+          "# of days the prodromal period lasts before the individual develops a rash"
```
Should be "average # of days..." if this is modeled as a stochastic rate instead of a fixed duration. Same for other periods below.

gvegayon replied:
It is average number of days.

damontoth commented on inst/models/shiny_measles.md:
```diff
@@ -5,7 +5,9 @@
+# Modeling Measles in Schools
 
-Using this app, you can simulate measles outbreaks and compare **how many fewer cases a quarantine procedure yields**. You can specify the number of people (population size), the number of initial cases, the proportion of vaccinated agents, and the duration in days. You can learn more about the model at <https://github.com/EpiForeSITE/epiworld-measles>.
+This model simulates measles outbreaks in schools and compares **how many fewer cases a quarantine procedure yields**. You can specify the number of people in the school (population size), the number of initial cases, the proportion of vaccinated agents, and the simulation duration in days. 
```
I suggest adding a brief description of "the quarantine procedure" here. Who is directed to to stay home from school and when.
```suggestion
This model simulates measles outbreaks in schools and compares **how many fewer cases a quarantine procedure yields**. You can specify the number of people in the school (population size), the number of students initially infected with measles (initial cases), the proportion of students who are vaccinated before the outbreak, and the simulation duration in days. 
```

damontoth commented on inst/models/shiny_measles.md:
```diff
@@ -5,7 +5,9 @@
-Using this app, you can simulate measles outbreaks and compare **how many fewer cases a quarantine procedure yields**. You can specify the number of people (population size), the number of initial cases, the proportion of vaccinated agents, and the duration in days. You can learn more about the model at <https://github.com/EpiForeSITE/epiworld-measles>.
+This model simulates measles outbreaks in schools and compares **how many fewer cases a quarantine procedure yields**. You can specify the number of people in the school (population size), the number of initial cases, the proportion of vaccinated agents, and the simulation duration in days. 
+
+Note that the model only simulates outbreaks within a single school and does not include transmissions that come into the school from external sources (e.g., other schools, the community, etc.). Learn more about the model at <https://github.com/EpiForeSITE/epiworld-measles>.
```
```suggestion
Note that the model only simulates outbreaks among students within a single school. It does not include transmissions from the students to people in other locations (e.g., other schools, households, the community, etc.) and does not include measles introductions to the school after the initial cases. Learn more about the model at <https://github.com/EpiForeSITE/epiworld-measles>.
```

damontoth commented on inst/models/shiny_measles.R:
```diff
@@ -367,69 +402,121 @@ measles_panel <- function(model_alt) {
+          slider_input_rate(
+            "measles", "Vaccination Improved Recovery", "0.5", input_label = "vax_improved_recovery"),
+          placement = "right",
+          "How much a vaccinated individual recovers faster than an unvaccinated individual"
```
```suggestion
          "How much faster a vaccinated infected individual recovers compared to an unvaccinated infected individual"
```

damontoth commented on inst/models/shiny_measles.R:
```diff
@@ -367,69 +402,121 @@ measles_panel <- function(model_alt) {
+            step    = 1
+          ),
+          placement = "right",
+          "# of days the rash lasts before the individual recovers"
```
They don't completely recover at the end of the rash, do they? Maybe just "Average number of days the rash lasts"

gvegayon replied:
It's average number of days. In the model is implemented as 1/# of days.

damontoth commented on NEWS.md:
In the table of results in the Outbreak Size section, I suggest changing the heading of the second column from "Probability" to "Probablity > Size"

apulsipher committed:
Apply suggestions from @damontoth code review

Co-authored-by: damontoth <Damon.Toth@hsc.utah.edu>

apulsipher committed:
change table column name to Probability > Size

apulsipher committed:
Merge branch '31-measles-updates' of github.com:UofUEpiBio/epiworldRShiny into 31-measles-updates

apulsipher commented:
@damontoth Thank you for your feedback! Using the suggestion feature made it really easy to apply your changes.

@gvegayon When you get a minute, can you look over the remaining comments from Damon? I've applied the suggestions I could, but there are a few that you should clarify before I make any changes.

gvegayon committed:
Updating the news and adapting the model to the latest version of epiworldR

gvegayon committed:
Solving conflicts

gvegayon commented:
I've update the NEWS, added a requirement to use epiworldR >= 0.8.2.0, and updated the model to reflect the renamed parameters. One thing I don't like about the UI, is the four panels:

![image](https://github.com/user-attachments/assets/25654158-0372-4fca-addf-cecedf5ebd80)

It took me work to understand what was going on. Perhaps we could just have two: with (blue) and without (red) quarantine, so each card has both numbers (cases + hospitalizations)? I looked for examples but was unable to find any matching what I describe. I'll try later

apulsipher commented:
> I've update the NEWS, added a requirement to use epiworldR >= 0.8.2.0, and updated the model to reflect the renamed parameters. One thing I don't like about the UI, is the four panels:
> 
> ![image](https://private-user-images.githubusercontent.com/893619/439331237-25654158-0372-4fca-addf-cecedf5ebd80.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDYwNDI2NzEsIm5iZiI6MTc0NjA0MjM3MSwicGF0aCI6Ii84OTM2MTkvNDM5MzMxMjM3LTI1NjU0MTU4LTAzNzItNGZjYS1hZGRmLWNlY2VkZjVlYmQ4MC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNDMwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDQzMFQxOTQ2MTFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1jZGVkYWRiODFmNDQzNjljNDFkZDc4ODMwZDZjMjlhMDY3ZWMzZmEyMTVkYjQ3ZTM5MWNjMzQ2MTE1ZDA5N2Y1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.dppDH6pptuw0gN3xhjLXhCmNiSB5e3YpWZJiIHyS-NY)
> 
> It took me work to understand what was going on. Perhaps we could just have two: with (blue) and without (red) quarantine, so each card has both numbers (cases + hospitalizations)? I looked for examples but was unable to find any matching what I describe. I'll try later

Maybe the value box UI element (those colored boxes) is the wrong element here. The value box is designed to highlight a single number and gets cluttered really fast when you try to do more than that. We're trying to highlight four values and, like you mentioned, compare two paired values. I think a table is better UI here, but possibly with colored cells. It'll be tighter and more conducive to comparison.

apulsipher committed:
Delete tooltip png and add epiworldR dev repo to DESCRIPTION

apulsipher committed:
Add Average # number of days to several tooltips

apulsipher committed:
Compress take home messages into two value boxes instead of four

apulsipher committed:
Make clear that quarantine and isolation days are not averages in shiny_measles.R

apulsipher committed:
Add (quarantine) label) to 95% CI in epicurve

apulsipher committed:
Merge branch '31-measles-updates' of github.com:UofUEpiBio/epiworldRShiny into 31-measles-updates

apulsipher committed:
Clean up outbreak table
